packages/react/src/useAction.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  actionOperation,
+  capitalize,
+  gadgetErrorFor,
+  GadgetRecord,
+  get,
+  hydrateRecord,
+  LimitToKnownKeys,
+  Select,
+} from "@gadgetinc/api-client-core";
+import { useMemo } from "react";
+import { CombinedError, useMutation, UseMutationResponse } from "urql";
+import { ActionFunction } from "./GadgetFunctions";
+import { OptionsType } from "./OptionsType";
+import { useStructuralMemo } from "./useStructuralMemo";
+
+/**
+ * React hook to fetch a Gadget record using the `findByXYZ` method of a given manager.
+ *
+ * @param action `findByXYZ` function from a Gadget manager that will be used
+ * @param params input params to the Gadget action
+ * @param options options for selecting the fields in the result
+ */
+export const useAction = <
+  OptionsT extends OptionsType, // currently necessary for Options to be a narrow type (e.g., `true` instead of `boolean`)
```

Comment:
Comment says it all. Without this, we don't get narrowed types for the options. In other words:

```
useFindOne(client.widget, { select: { name: true } });
```

Will give us `name: boolean` instead of `name: true`. I don't completely understand the why behind this, but it appears to work! Hoping one day we can get rid of it.

Reply:
Yep, happy to change the name to something more useful like that 🚀